src/text/parsers/float.rs

```diff
@@ -0,0 +1,133 @@
+use nom::{IResult, Parser};
+use crate::text::TextStreamItem;
+use nom::sequence::{terminated, preceded, tuple};
+use nom::branch::alt;
+use crate::text::parsers::stop_character;
+use nom::combinator::{map_res, opt, recognize, map};
+use nom::bytes::streaming::{tag};
+use nom::character::streaming::one_of;
+use crate::text::parsers::numeric_support::{digits_before_dot, floating_point_number, exponent_digits};
+use std::num::ParseFloatError;
+use std::str::FromStr;
+
+/// Matches the text representation of a float value and returns the resulting [f64]
+/// as a [TextStreamItem::Float].
+pub(crate) fn parse_float(input: &str) -> IResult<&str, TextStreamItem> {
```

Comment:
Mostly a curiosity comment: there are similarities between float parsing and double parsing, could we, in the future, refactor a bit or does `nom` make that difficult?

Reply:
Very probably; I would need to carefully review which functions are used where. There's probably some room for consolidation.